bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java

```diff
@@ -845,7 +845,7 @@ void addEntry(final long ledgerId, byte[] masterKey, final long entryId, Referen
             // usually checked in writeAndFlush, but we have extra check
             // because we need to release toSend.
             errorOut(completionKey);
-            ReferenceCountUtil.release(toSend);
+            ReferenceCountUtil.release(request);
```

Comment:
Does this also result in the correct behavior when `useV2WireProtocol == false`?

It doesn't look consistent how the V3 protocol is currently handled since there's no retained duplicate and the backing Netty ByteBuf isn't released.
In the case of V3 protocol, the body might consist of com.google.protobuf.NioByteString instances that wrap the Nio ByteBuffer instances backed by Netty ByteBuf instances which also control the lifecycle of the Nio ByteBuffer instances.

Reply:
It looks like the concern about the NioBuffer lifecycle was raised by @sijie already in the original PR review: https://github.com/apache/bookkeeper/issues/791#issuecomment-383237059 .

Reply:
Added `release` at these locations

Reply:
> Does this also result in the correct behavior when useV2WireProtocol == false?

No memory leak when `useV2WireProtocol` is `false`, but the error `io.netty.util.IllegalReferenceCountException: refCnt: 0, increment: 1` will also occur.